internal/cmd/generate.go

```diff
@@ -135,3 +133,53 @@ func Generate(dir string, stderr io.Writer) (map[string]string, error) {
 	}
 	return output, nil
 }
+
+func parse(name, dir string, sql config.SQL, combo config.CombinedSettings, parserOpts dinosql.ParserOpts, stderr io.Writer) (dinosql.Generateable, bool) {
+	switch sql.Engine {
+	case config.EngineMySQL:
+		// Experimental MySQL support
+		q, err := mysql.GeneratePkg(name, sql.Schema, sql.Queries, combo)
+		if err != nil {
+			fmt.Fprintf(stderr, "# package %s\n", name)
+			if parserErr, ok := err.(*dinosql.ParserErr); ok {
+				for _, fileErr := range parserErr.Errs {
+					printFileErr(stderr, dir, fileErr)
+				}
+			} else {
+				fmt.Fprintf(stderr, "error parsing schema: %s\n", err)
+			}
+			return nil, true
+		}
+		return q, false
+
+	case config.EnginePostgreSQL:
+		c, err := dinosql.ParseCatalog(sql.Schema)
+		if err != nil {
+			fmt.Fprintf(stderr, "# package %s\n", name)
+			if parserErr, ok := err.(*dinosql.ParserErr); ok {
+				for _, fileErr := range parserErr.Errs {
+					printFileErr(stderr, dir, fileErr)
+				}
+			} else {
+				fmt.Fprintf(stderr, "error parsing schema: %s\n", err)
+			}
+			return nil, true
+		}
+
+		q, err := dinosql.ParseQueries(c, sql.Queries, parserOpts)
+		if err != nil {
+			fmt.Fprintf(stderr, "# package %s\n", name)
+			if parserErr, ok := err.(*dinosql.ParserErr); ok {
+				for _, fileErr := range parserErr.Errs {
+					printFileErr(stderr, dir, fileErr)
+				}
+			} else {
+				fmt.Fprintf(stderr, "error parsing queries: %s\n", err)
+			}
+			return nil, true
+		}
+		return &kotlin.Result{Result: q}, false
```

Comment:
Always returning a Kotlin result here seems strange. Is this just so that you can't generate MySQL code for Kotlin?

Reply:
Yeah, that, and me trying to minimize my changes to the generation code in fear of nasty merge conflicts. Happy to change if you can suggest a more elegant way of doing it. (check the lang here?)

Reply:
I think this is okay for now, but something we'll probably want to change once we land a third language backend.